Controllers/CartController.cs

```diff
@@ -177,14 +177,59 @@ public CartController(BangazonContext ctx)
                 openOrder.DateCompleted = DateTime.Now;
                 context.Order.Update(openOrder);
                 await context.SaveChangesAsync();
-                return RedirectToAction("Index","Cart");
+                return RedirectToAction("Confirmation","Cart");
 
             }
             catch (DbUpdateException)
             {
                 throw;
             }   
         }
+        public async Task<IActionResult> Confirmation()
+        {
+            User user = ActiveUser.Instance.User;
+            int? userId = user.UserId;
+
+            if (userId == null)
+            {
+                return Redirect("ProductTypes");
+            }
+
+            Order CompleteOrder = await(
+                from order in context.Order
+                where order.UserId == userId && order.DateCompleted != null 
+                select order).SingleOrDefaultAsync();
+                
+            if (CompleteOrder == null)
```

Comment:
I'm not quite sure we need these order checks, but I think we could combine into 1:
`if (CompleteOrder == null || CompleteOrder.UserId != userId)`